pkg/combustion/network.go

```diff
@@ -0,0 +1,151 @@
+package combustion
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"github.com/suse-edge/edge-image-builder/pkg/template"
+	"go.uber.org/zap"
+)
+
+const (
+	nmcExecutable           = "nmc"
+	nmcConfigDir            = "config"
+	networkConfigDir        = "network"
+	networkConfigScriptName = "configure-network.sh"
+)
+
+//go:embed templates/configure-network.sh.tpl
+var configureNetworkScript string
+
+// Configures the network component if enabled.
+//
+//  1. Generates network configurations
+//  2. Copies the NMC executable
+//  3. Writes the configuration script
+//
+// Example result file layout:
+//
+//	combustion
+//	├── network
+//	│   ├── config
+//	│   │   ├── node1.example.com
+//	│   │   │   ├── eth0.nmconnection
+//	│   │   │   └── eth1.nmconnection
+//	│   │   ├── node2.example.com
+//	│   │   │   └── eth0.nmconnection
+//	│   │   ├── node3.example.com
+//	│   │   │   ├── bond0.nmconnection
+//	│   │   │   └── eth1.nmconnection
+//	│   │   └── host_config.yaml
+//	│   └── nmc
+//	└── configure-network.sh
+func configureNetwork(ctx *image.Context) ([]string, error) {
+	networkDir, err := generateComponentPath(ctx, networkConfigDir)
+	if err != nil {
+		return nil, fmt.Errorf("generating network path: %w", err)
+	}
+
+	if networkDir == "" {
+		// Networking is not configured, exit early
+		return nil, nil
+	}
+
+	zap.L().Info("Configuring network component...")
```

Comment:
What do you think about moving this above the punch out check and then logging in there something like "No network configuration detected"?

It means the output is chattier since it'll show things that aren't relevant to the image in question, but it also makes it more clear what EIB is doing.

Alternatively, do we want the policy to be that we log to debug when we punch out early, so that way we can still see that's what happened but only for debugging cases?

Reply:
BTW, I mean this as part of the general logging policy we were talking about for these sorts of Info messages.

Reply:
I agree on there being no right/wrong answer. I also had a similar thought on the chattier, how it would implicitly teach the user about other options.

My vote is for chattier; we log each component as it's configured and always have some sort of closure to it, either "Finished configuring" or "Skipping since there is no configuration".